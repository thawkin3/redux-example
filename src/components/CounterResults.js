import React from 'react';
import { connect } from 'react-redux';
import './CounterResults.css';

export const CounterResults = ({ clickCount }) => (
    <p>Counter value: {clickCount}</p>
);

function mapStateToProps({ clickCount }) {
    return { clickCount };
}

CounterResults.displayName = 'CounterResults';

export default connect(mapStateToProps, null)(CounterResults);
