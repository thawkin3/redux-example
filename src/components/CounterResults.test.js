import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { CounterResults } from './CounterResults';

let wrapper;
let props;

describe('CounterResults', () => {
    beforeEach(() => {
        props = { clickCount: 150 };
        wrapper = shallow(<CounterResults {...props} />);
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CounterResults {...props} />, div);
    });

    it('matches the snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders only one CounterResults', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('renders a paragraph element', () => {
        expect(wrapper.find('p').length).toEqual(1);
    });

    it('shows the correct click count based on the props', () => {
        expect(wrapper.find('p').prop('children')).toContain(150);
    });
});
