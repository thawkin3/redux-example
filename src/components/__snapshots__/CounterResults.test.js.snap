// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterResults matches the snapshot 1`] = `
<p>
  Counter value: 
  150
</p>
`;
